lib/ai/providers/openai-provider.ts

```
@@ -0,0 +1,31 @@
+
+  async generateStreamingResponse(prompt: string): Promise<ReadableStream> {
+    throw new Error("Method not implemented. Could not process: " + prompt);
+  }
+
+  async generateEmbedding(value: string): Promise<number[]> {
+    try {
+      const model = this.client.embedding(AI_MODELS.OPENAI.EMBEDDING);
+      const response = await model.doEmbed({ values: [value] });
+      return response.embeddings.map(x => x as unknown as number);
+    } catch (err) {
+      throw err;
```

Biome (1.9.4)

[error] The catch clause that only rethrows the original error is useless.

An unnecessary catch clause can be confusing.
Unsafe fix: Remove the try/catch clause.


(lint/complexity/noUselessCatch)